core/libp2p/multi/multibase/multibase_impl.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/multi/multibase/multibase_impl.hpp"
+
+#include <optional>
+#include <unordered_map>
+
+#include "libp2p/multi/multibase/codecs/base16.hpp"
+#include "libp2p/multi/multibase/codecs/base58.hpp"
+#include "libp2p/multi/multibase/codecs/base64.hpp"
+
+namespace {
+  using namespace libp2p::multi;
+  using namespace libp2p::multi::detail;
+
+  /**
+   * Get the encoding by a character
+   * @param ch of encoding
+   * @return related encoding, if character stands for one of them, none
+   * otherwise
+   */
+  std::optional<Multibase::Encoding> encodingByChar(char ch) {
+    switch (ch) {
+      case 'f':
+        return Multibase::Encoding::kBase16Lower;
+      case 'F':
+        return Multibase::Encoding::kBase16Upper;
+      case 'Z':
+        return Multibase::Encoding::kBase58;
+      case 'm':
+        return Multibase::Encoding::kBase64;
+      default:
+        return {};
+    }
+  }
+
+  using EncodeFuncType = decltype(encodeBase64);
+  using DecodeFuncType = decltype(decodeBase64);
+
+  /// all available codec functions
+  const std::unordered_map<Multibase::Encoding,
```

Comment:
make it static

Reply:
Well, as it seems, variables in anonymous namespaces are some kind of static by default, but OK, I will explicitly add the keyword